### PR TITLE
support for custom data in vmss

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Just adding support for custom data in VMSS machine scaleset

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine_scaleset

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py [hunk 1]
```diff
@@ -206,6 +206,11 @@
             - A list of Availability Zones for your virtual machine scale set
         type: list
         version_added: "2.8"
+    custom_data:
+        description:
+            - Custom data for using cloud-init for your VM.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py]: it is a file name? or url? or file content?

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py [hunk 2]
```diff
@@ -498,6 +507,9 @@ def exec_module(self, **kwargs):
             # Set default location
             self.location = resource_group.location
 
+        if self.custom_data:
+            self.custom_data = to_native(base64.b64encode(to_bytes(self.custom_data)))
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py:514]: from code, it's  script content, but i think url or filename should be supported for long script.

**Review by yungezz — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py]: it's content, in rest api it's base64 encoded content.
so I follow the same rules as in vm (vm has the same option).
I think I will add entire description

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py (hunk 2 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py:514]: it's limited to 16384 bytes.
users can use Ansible lookups to read the content from file or uri.
i don't think we should make it more complicated here.

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py:514]: make sense

**Review by yungezz — APPROVED** [review]

**Comment by zikalino** [comment]
@mattclay seems like I have some weird problem here. resource group can't be found.....

**Comment by mattclay** [comment]
@zikalino We were having issues running Azure tests earlier. I've restarted the failed tests.

**Pull request merged**